app/components/participant-creator/component.js

```diff
@@ -99,28 +101,42 @@ export default Ember.Component.extend({
         },
         addExtraField() {
             var next = this.get('nextExtra');
-            this.get('extra').pushObject({
-                key: next,
-                value: null
-            });
+            var fieldExists = false;
+            // Do not allow users to add duplicate studyId field
+            if (next === 'studyId') {
+                fieldExists = true;
+            }
+            for (var item of this.get('extra')) {
```

Comment:
Could this loop be written more concisely as `Object.keys(extra).includes(next)`?

That would let you consolidate lines 104-114 down into a short conditional check for line 115.